File: home_assignments/csharp/TicketSeller/server/implementations/UserService.cs
using System;
using log4net;
using model.entities;
using model.validators;
using model.validators.interfaces;
using persistence.interfaces;
using services;
using services.interfaces;

namespace server.implementations;

public class UserService : IUserService
{
    private static readonly ILog Logger = LogManager.GetLogger("UserService");


    private readonly IUserValidator _userValidator;
    private readonly IUserRepository _userRepository;

    /**
     * constructs a user service based on given validator and repository
     *
     * @param userValidator  - said validator
     * @param userRepository - said repository
     */
    public UserService(IUserValidator userValidator, IUserRepository userRepository)
    {
        this._userValidator = userValidator;
        this._userRepository = userRepository;
        Logger.InfoFormat("Instantiated UserService with repo:{0} and validator:{1}", userRepository, userValidator);
    }

    public void Login(string username, string password)
    {
        Logger.InfoFormat("trying to login with username:{0} and password:{1}", username, password);
        try {
            _userValidator.ValidateForLogin(username, password);
        } catch (ValidationException e) {
            Logger.Error(e);
            throw e;
        }
        User user = _userRepository.FindById(username);
        Boolean toReturn = user != null && user.Password.Equals(password);
        if(!toReturn){
            throw new ServiceException("No user with given credentials");
        }
        Logger.Info("Exiting login");
    }
}

File: home_assignments/java/TicketSeller/TicketSellerServer/src/main/java/ticketseller/server/UserService.java
package ticketseller.server;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ticketseller.model.entities.User;
import ticketseller.model.validators.interfaces.IUserValidator;
import ticketseller.persistence.interfaces.IUserRepository;
import ticketseller.model.validators.ValidationException;
import ticketseller.services.ServiceException;
import ticketseller.services.interfaces.IUserService;

public class UserService implements IUserService {

    private static final Logger logger = LogManager.getLogger();

    private final IUserValidator userValidator;
    private final IUserRepository userRepository;

    /**
     * constructs a user service based on given validator and repository
     *
     * @param userValidator  - said validator
     * @param userRepository - said repository
     */
    public UserService(IUserValidator userValidator, IUserRepository userRepository) {
        this.userValidator = userValidator;
        this.userRepository = userRepository;
        logger.info("Instantiated UserService with repo:{} and validator:{}", userRepository, userValidator);
    }

    @Override
    public void login(String username, String password) throws ServiceException {
        logger.traceEntry("trying to login with username:{} and password:{}", username, password);
        try {
            userValidator.validateForLogin(username, password);
        } catch (ValidationException e) {
            logger.error(e);
            throw new ServiceException(e.getMessage());
        }
        User user = userRepository.findByID(username);
        boolean toReturn = user != null && user.getPassword().equals(password);
        if(!toReturn){
            throw new ServiceException("No user with given credentials");
        }
        logger.traceExit();
    }
}
